MRN-Code/coinstac
# Pipeline--UI--Pipeline-Server Integration

* **Problem:**
   1. Pipeline code is not reachable from UI
* **Solution:**
   1. Integrate pipeline into UI via client calls to `coinstac-pipeline`
   1. Consortia GraphQL call to start remote pipeline and notify other clients via subscriptions on `run` table
   1. Create pipeline server to listen for new pipeline calls from GraphQL and to write results back to graphql-server
   1. Make necessary changes to UI schema to mesh with pipeline schema
      1. Changes so far limited to those necessary for single step decentralized pipeline
      1. More changes yet to make concerning source mapping for step inputs 
   1. Pipeline start & complete notifications
* **Tests:**
   1. Login with `test` user.
   1. Click on Consortia nav item to reveal consortia list including one item with a *Start Pipeline* button.
   1. Click the button to start a pipeline and await results, both of which should throw notifications.
